Open Judge System/Web/OJS.Web/Areas/Api/Controllers/ResultsController.cs

```diff
@@ -168,25 +168,27 @@ public JsonResult GetAllUserResultsPercentageByForContest(string apiKey, int? co
                 return this.Json(new ErrorMessageViewModel("Invalid API key"), JsonRequestBehavior.AllowGet);
             }
 
+            var contestMaxPoints = this.data
```

Comment:
This will make more than one db call.

You can try something like that:
```
this.data.Problems
    .All()
    .Where(p => p.ContestId == contestId.Value)
    .Select(p => (double) p.MaximumPoints)
    .DefaultIfEmpty(1)
    .Sum();
```